test/hotspot/jtreg/compiler/c2/irTests/TestVectorizationMismatchedAccess.java

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright (c) 2023, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2.irTests;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Utils;
+import jdk.test.whitebox.WhiteBox;
+import jdk.internal.misc.Unsafe;
+import java.util.Random;
+import java.util.Arrays;
+import java.nio.ByteOrder;
+
+/*
+ * @test
+ * @bug 8300258
+ * @key randomness
+ * @requires (os.simpleArch == "x64") | (os.simpleArch == "aarch64")
```

Comment:
Does this require `@requires compiler.c2`? What is the expectation if this (or any IR) test runs on GraalVM?
cc @chhagedorn